gnocchi/indexer/sqlalchemy.py

```diff
@@ -1101,8 +1194,50 @@ def _get_history_result_mapper(self, session, resource_type):
         s1 = sqlalchemy.select(history_cols.values())
         s2 = sqlalchemy.select(resource_cols.values())
         if resource_type != "generic":
+            engine = session.connection()
+
+            resource_history_filters_names = [
```

Comment:
You could use the sqlalchemy to get the Resource history table columns instead of hardcoding them
```suggestion
            # Get the available columns for ResourceHistory table
            resource_history_cols = list(
                map(lambda column: column.name, sqlalchemy.inspect(
                    ResourceHistory).columns))
```